scripts/hostcfgd

```diff
@@ -550,12 +561,12 @@ class Iptables(object):
                 as a new rule even if it is the same as an existing one. Check this and
                 do nothing if rule exists
                 '''
-                ret = subprocess.call(cmd, shell=True)
+                ret = subprocess.call(cmd)
                 if ret == 0:
                     syslog.syslog(syslog.LOG_INFO, "{} rule exists in {}".format(ip, chain))
                 else:
                     # Modify command from Check to Append
-                    iptables_cmds.append(cmd.replace("check", "append"))
+                    iptables_cmds.append([word if word != "--check" else "--append" for word in cmd])
```

Comment:
>cmd [](http://example.com/codeflow?start=96&length=3)

Strictly speaking, you changed behavior.
Suggest loop through cmd, replace in each element.